### PR TITLE
Remove IDisposable from AuthenticationProvider base class

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: authentication/src/AuthenticationProvider.cs [hunk 1]
```diff
@@ -1,33 +1,16 @@
...
-        /// <summary>
-        /// Releases the unmanaged resources and disposes of the managed resources used by the invoker.
+        /// Gets the registration Id used during device enrollment.
         /// </summary>
-        public void Dispose()
-        {
-            Dispose(true);
-            GC.SuppressFinalize(this);
-        }
+        public abstract string GetRegistrationId();
```
Comment (drwill-ms) [inline authentication/src/AuthenticationProvider.cs:14]: Fixed case of Id

File: security/tpm/samples/AuthenticationProviderTpmSimulator/AuthenticationProviderTpmSimulator.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ namespace Microsoft.Azure.Devices.Provisioning.Security.Samples
     /// This code is provides as a sample to enable provisioning on hardware without an actual hardware TPM device and
     /// provides no real security.
     /// </summary>
-    public class AuthenticationProviderTpmSimulator : AuthenticationProviderTpm
+    public class AuthenticationProviderTpmSimulator : AuthenticationProviderTpm, IDisposable
```
Comment (drwill-ms) [inline security/tpm/samples/AuthenticationProviderTpmSimulator/AuthenticationProviderTpmSimulator.cs:19]: Only this instance was disposable, so now it inherits the interface directly

File: security/tpm/src/AuthenticationProviderTpmHsm.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@ namespace Microsoft.Azure.Devices.Provisioning.Security
     /// <summary>
     /// The Provisioning Authentication Client implementation for TPM.
     /// </summary>
-    public class AuthenticationProviderTpmHsm : AuthenticationProviderTpm
+    public class AuthenticationProviderTpmHsm : AuthenticationProviderTpm, IDisposable
```
Comment (drwill-ms) [inline security/tpm/src/AuthenticationProviderTpmHsm.cs:14]: Only this instance was disposable, so now it inherits the interface directly

File: security/tpm/samples/AuthenticationProviderTpmSimulator/AuthenticationProviderTpmSimulator.cs [hunk 4]
```diff
@@ -120,16 +120,13 @@ public static void StartSimulatorProcess()
             simulatorProcess.Start();
         }
 
-        protected override void Dispose(bool disposing)
+        public void Dispose()
```
Comment (timtay-microsoft) [inline security/tpm/samples/AuthenticationProviderTpmSimulator/AuthenticationProviderTpmSimulator.cs:123]: xml header?

**Review by timtay-microsoft — APPROVED** [review]

**Pull request merged**